application/Synchronization/AmazonSynchronizer.cs

```diff
@@ -125,10 +139,14 @@ private void ExecuteAndWait(Action action, int delayInMs = 1000)
             }
             catch (Exception ex)
             {
-                _logger.WriteEntry($"@ExecuteAndWait : {ex.GetLastErrorMessage()}", LoggingLevel.Error, _updatesCount);
+                _telemetryClient.TrackException(ex, new Dictionary<string, string>()
+                {
+                    {"Source", "ExecuteAndWait" },
+                    {"Update#" ,_updatesCount.ToString()}
+                });
             }
 
-            _logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
+            //_logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
```

Comment:
There is a way to record timings in `ApplicationInsights`, I don't remember what the name of the method is but one of them takes a duration I believe.  Would be nice to get this recorded as such.